app/models/academic_cycle.rb

```diff
@@ -1,10 +1,13 @@
 # frozen_string_literal: true
 
 class AcademicCycle < ApplicationRecord
+  has_many :funding_methods
+
   validates :start_date, :end_date, presence: true
   validate :start_date_before_end_date
 
   scope :trainees_filter, -> { order(start_date: :desc).limit(3) }
+  scope :for_date, ->(date) { where("end_date >= :date AND start_date <= :date", date: date) }
```

Comment:
This should only ever return one result. I think we should make it consistent with `.for_year` and implement it as a method that returns a single academic cycle rather than a scope.